### ISSUE TITLE
🐛 BUG: Astro packages don't properly define their dependencies (Yarn PnP incompatibilty)

### ISSUE THREAD

**Issue by andreialecu** [withastro/astro#1676]
### What package manager are you using?

yarn 3.1

### What operating system are you using?

Mac

### Describe the Bug

I've been experimenting with Astro and Yarn 3 in PnP mode.

PnP mode is strict about packages requiring dependencies, and it does not allow them to be hijacked from other packages.

Astro is doing some things wrong in this regard, for example it uses `fast-glob` but it's nowhere in `package.json`. I suspect the `pnpm` package manager will also have problems with this.

### Steps to Reproduce

```
yarn create astro --template framework-react
yarn set version berry
yarn
yarn dev
```

```
➜ yarn dev       
(node:91510) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Error: astro tried to access fast-glob, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: fast-glob (via "fast-glob/package.json")
Required by: astro@npm:0.20.12 (via /.../.yarn/cache/astro-npm-0.20.12-5a08c46146-1be4d53e64.zip/node_modules/astro/dist/check.js)
```

`fast-glob` is indeed imported here:
https://github.com/snowpackjs/astro/blob/b1b564d03d09cee63e072397dfb2ab1a1f59b346/packages/astro/src/check.ts#L5

But it's not defined in any `package.json`.

This is a correctness issue. See: https://yarnpkg.com/advanced/rulebook/#packages-should-only-ever-require-what-they-formally-list-in-their-dependencies

**Comment by drwpow** [comment]
Minor update on this: I’ve been testing `astro@next` (0.21 beta) locally using the latest version of pnpm with no issues. I believe pnpm support should be fixed with the upcoming release. Unknown about Yarn 3.

**Comment by FredKSchott** [comment]
Confirmed issues in astro@next on yarn `berry` mode. I think pnpm still has the same issues, mainly around our renderer loading system. We can aim to prioritize this post v0.21.0 release

**Comment by tony-sull** [comment]
Just confirmed I'm still seeing this with yarn `berry`

Looks like [https://github.com/yarnpkg/berry/pull/2161](yarnpkg/berry#2161) is merged in now.  I'm not super familiar with the inner workings of the different package managers, is this something we should fix on our end or is it blocked on a `berry` issue?

**Comment by FredKSchott** [comment]
Astro monorepo is now on pnpm, so this should be mostly resolved. However, `shamefully-hoist` is still recommended.

**Comment by MartijnHarmenzon** [comment]
I am still not able to use yarn berry. When using the following steps:
1. yarn set version berry
2. yarn create astro

This wil resolve in the following error:
> Error: chalk tried to access #ansi-styles, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Any ideas?

**Comment by MrAlex94** [comment]
@FredKSchott, I think this has been incorrectly closed. `pnpm` is separate from yarn's Plug'n'Play (`pnp`), which this thread is talking about.

I have this issue, and need to add extra dependencies with `yarn`.

**Comment by bluwy** [comment]
@MrAlex94 we recently fully support projects without shamefully-hoist, which is a common issue with strict dependency installations like pnpm and yarn pnp. Are there anything else specific that we didn't cover? I might be better to create a separate issue for.
